### ISSUE TITLE
[baget.io] Add Google Analytics

### ISSUE THREAD

**Issue by loic-sharma** [loic-sharma/BaGet#360]
Configured by CD pipeline. If not configured, GA shouldn't be included.

**Comment by loic-sharma** [comment]
Hey @DMW007, why did you down vote this? Just curious, I'd appreciate any feedback!

**Comment by DMW007** [comment]
Hi @loic-sharma,

while collecting usage statistics is reasonable and verifiable, using proprietary Google service is not. Especially when it comes to privacy. Google is a company that collects as many data as possible for selling them and also giving them to the US government, maybe also others. So using GA is not compatible with privacy. 

If you want to respect the users privacy, please look for open solutions like [Matomo](https://matomo.org/) and host them in a country with strong data protection laws.

**Comment by loic-sharma** [comment]
That makes sense. In the interest of time, I'd like to avoid hosting my own solution. I'd also prefer a solution that is free, or can be run on Azure (as I have a free Azure subscription). What do you think of [Azure Application Insights' client-side telemetry](https://docs.microsoft.com/en-us/azure/azure-monitor/app/javascript#sending-telemetry-to-the-azure-portal)? Here is the telemetry it collects:

> * Uncaught exceptions in your app, including information on
>     * Stack trace
>     * Exception details and message accompanying the error
>     * Line & column number of error
>    * URL where error was raised
> * Network Dependency Requests made by your app XHR and Fetch (fetch collection is disabled by default) requests, include information on
>     * Url of dependency source
>     * Command & Method used to request the dependency
>     * Duration of the request
>     * Result code and success status of the request
>     * ID (if any) of user making the request
>     * Correlation context (if any) where request is made
> * User information (for example, Location, network, IP)
> * Device information (for example, Browser, OS, version, language, resolution, model)
> * Session information

From my understanding, Microsoft does not share any telemetry collected in Application Insights. Does this seem like a decent alternative?

**Comment by DMW007** [comment]
Hosting on third party cloud services always results in giving up control. Especially Azure has bad reputation in Germany as well as other EU countries, like any US based solution because of intelligence spying and the US laws that doesn't propetct users data (especially when they're not living in the USA). Azure isn't free for nothing, the same like Windows 10 (which is de facto free: Could be activated with nearly any key from previous versions or $4 keys from ebay, it runs even without activation). 

So for respecting privacy, you need at least hosting in the EU. Don't think that this could be realized for free (at least not without paying by data). But a compromise could be reputated EU hosting provider that offers webspace. This shouldnt cost more than a few dollars per month. Installing Matomo on a webspace isn't a big deal. I run it myself (on Docker, I have a server where also other tools like BaGet were hosted) and also did installations without Docker in the past.

**Comment by loic-sharma** [comment]
I agree with you, but I'd like to balance privacy concerns with convenience. Launching this website is already a huge undertaking, so I'd like to keep it simple without compromising on my own morals. At this time, I believe that Application Insights is an acceptable solution **so long as I can prevent Application Insights from collecting users' personal information** (like IP addresses, network, and whatever else could be used for tracking). I've reviewed Application Insight's documentation, and these claim that:

1. Microsoft will not access Application Insights telemetry without permission ([source](https://docs.microsoft.com/en-us/azure/azure-monitor/app/data-retention-privacy#how-secure-is-my-data))
1. Microsoft only uses Application Insights data only to provide the service to Application Insights customers ([source](https://docs.microsoft.com/en-us/azure/azure-monitor/app/data-retention-privacy#what-does-microsoft-do-with-the-information-my-app-sends-to-application-insights))

> Azure isn't free for nothing, the same like Windows 10 (which is de facto free: Could be activated with nearly any key from previous versions or $4 keys from ebay, it runs even without activation).

My "free Azure subscription" statement in my previous comment was a little misleading; as a Microsoft employee I do not have to pay for the first $150 I spend on Azure each month.
